cms/jinja2/templates/pages/release_calendar/release_calendar_page.html

```diff
@@ -1,47 +1,48 @@
 {% extends "templates/base_page.html" %}
 
-{% from "components/table-of-contents/_macro.njk" import onsTableOfContents %}
 {% from "components/hero/_macro.njk" import onsHero %}
+{% from "components/table-of-contents/_macro.njk" import onsTableOfContents %}
 
 {% block header_area %}
     {%- set hero_topic = _("Release") -%}
+    {%- set release_date_label = _("Release date:") -%}
+    {%- set next_release_date_label = _("Next release date:") -%}
     {%- set breadcrumbs_label = _("Breadcrumbs") -%}
 
-    {% if page.release_date or page.release_text %}
-        {%- set release_date_label = _("Release date:") -%}
-        {# fmt:off #}
-        {%- set release_date_html =
-            {
-                "term": release_date_label,
-                "descriptions": [
-                    {
-                        "description": page.release_date|ons_date("DATETIME_FORMAT") or page.release_date_text
-                    }
-                ]
-            }
-        -%}
-        {# fmt:on #}
-    {% else %}
-        {%- set release_date_html = "" -%}
-    {% endif %}
-
-    {% if page.next_release_date or page.next_release_text %}
-        {%- set next_release_date_term = _("Next release:") -%}
-        {# fmt:off #}
-        {%- set next_release_html =
-            {
-                "term": next_release_date_term,
+    {# fmt:off #}
+    {% set release_date_details = [
+        {
+        "type": "release_date",
+        "label": release_date_label,
+        "value": page.release_date,
+        "text": page.release_date_text
+        },
+        {
+        "type": "next_release_date",
+        "label": next_release_date_label,
+        "value": page.next_release_date,
+        "text": page.next_release_date_text
+        }
+    ] %}
```

Comment:
How about

```suggestion
    {%- set release_date_details = [
        {
            "label": release_date_label,
            "value": page.release_date_text or page.release_date|ons_date("DATETIME_FORMAT"),
        },
        {
            "label": next_release_date_label,
            "value": page.next_release_date_text or page.next_release_date|ons_date("DATETIME_FORMAT"),
        }
    ] -%}
```

since type is not used, and we do a valur or text anyway.

Then:

```jinja2
    {%- set release_dates_list = [] -%}
    {%- for release_date in release_date_details -%}
        {# fmt:off #}
        {% if release_date.value %}
            {# fmt:off #}
            {%- do release_dates_list.append({
                "term": release_date.label,
                "descriptions": [{"description": release_date.value}]
            }) -%}
            {# fmt:on #}
        {% endif %}
    {%- endfor -%}
```

Reply:
Went with this in: [902abf5](https://github.com/ONSdigital/dis-wagtail/pull/158/commits/902abf5e7ab80eb6a8bdbef68389a07aba335292)

In this commit I also included the next release dates in the templates for provisional, confirmed and published releases. Let's keep the convo about this part in this thread: https://github.com/ONSdigital/dis-wagtail/pull/158#discussion_r2063907359